tests/categorical/test_contingency.py

```diff
@@ -0,0 +1,315 @@
+    table2 = match.make_table(simple_forecast, simple_obs, event_threshold=1.3)
+    counts = table.get_counts()
+    actual_table = table.get_table()
+
+    # Test event tables creation matches the stored tables
+    fcst_events, obs_events = match.make_event_tables(simple_forecast, simple_obs)
+    fcst_events2, obs_events2 = match.make_event_tables(simple_forecast, simple_obs, event_threshold=1.3)
+    xr.testing.assert_equal(fcst_events, table.forecast_events)
+    xr.testing.assert_equal(fcst_events2, table.forecast_events)
+    xr.testing.assert_equal(obs_events, table.observed_events)
+    xr.testing.assert_equal(obs_events2, table.observed_events)
+    xr.testing.assert_equal(table.forecast_events, table2.forecast_events)
+    xr.testing.assert_equal(table.observed_events, table2.observed_events)
+
+    # Confirm values in the contingency table are correct
+    assert counts["tp_count"] == 9
+    assert counts["tn_count"] == 6
+    assert counts["fp_count"] == 2
+    assert counts["fn_count"] == 1
+    assert counts["total_count"] == 18
+
+    # Confirm that the xarray table object has the correct counts too
+    assert actual_table.sel(contingency="tp_count") == 9
+    assert actual_table.sel(contingency="tn_count") == 6
+    assert actual_table.sel(contingency="fp_count") == 2
+    assert actual_table.sel(contingency="fn_count") == 1
+    assert actual_table.sel(contingency="total_count") == 18
+
+    # Confirm calculations of metrics are correct
+    assert table.accuracy() == (9 + 6) / 18
+    assert table.probability_of_detection() == 9 / (9 + 1)
+    assert table.false_alarm_rate() == 2 / (2 + 6)
+    assert table.threat_score() == 9 / (9 + 2 + 6)
+    assert table.frequency_bias() == (9 + 2) / (9 + 1)
+    assert table.hit_rate() == 9 / (9 + 1)
+    assert table.probability_of_detection() == 9 / (9 + 1)
+    assert table.success_ratio() == 9 / (9 + 2)
+    assert table.specificity() == 6 / (6 + 2)
+
+    # These methods are redirects to eachother
+    assert table.critical_success_index() == table.threat_score()
+    assert table.hit_rate() == table.probability_of_detection()
+    assert table.sensitivity() == table.hit_rate()
+    assert table.false_alarm_rate() == table.probability_of_false_detection()
+    assert table.frequency_bias() == table.bias_score()
+
+    peirce_component_a = 9 / (9 + 1)
+    peirce_component_b = 1 / (1 + 6)
+    peirce_expected = peirce_component_a - peirce_component_b
+    assert table.peirce_skill_score() == peirce_expected
+
+def test_dimension_broadcasting():   
+    '''
+    Confirm that dimension broadcasting is working, through an example where a time
+    dimension is introduced to the forecast object. Each element of the time dimension
+    should be compared to the observation, and the final accuracy should have an
+    accuracy score at each time.
+
+    The actual calculated accuracy at each time is not tested here, this test is about
```

Comment:
Okay, it would be good to create an issue to add an issue to check that checks the output of at least one method (e.g., accuracy) is correct